[v1.5][W14-B4] CelebManager

muruges95 opened the pull request:
Start of pull request. Added documentation on user stories, use cases, NFR etc. in our Developer Guide for the Week 7 requirements. Will update with our individual enhancements to the different components of the address-book. The name of the addressbook will be Celeb Scheduler, as mentioned in the title. Unclear if we are supposed to put this name in the Developer Guide as well, but for now that has not been done.

Other team members (so they get notified): @Damienskt @WJY-norainu @tzerbin

muruges95 commented:
As of now this is our v1.0 of our app.

kowshik-sundararajan commented on docs/UserGuide.adoc:
```diff
@@ -11,7 +11,7 @@ ifdef::env-github[]
 :note-caption: :information_source:
 endif::[]
-:repoURL: https://github.com/se-edu/addressbook-level4
+:repoURL: https://github.com/CS2103W14-B4/addressbook-level4
```
Remember to change the repo name to `main`

kowshik-sundararajan commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -67,4 +69,24 @@ public boolean equals(Object other) {
+    */
+    private String getColorForTags(String tagName) {
+        /**
+         * we use the hash code of the tag name to generate a random color, so that the color remain consistent
```
Could you rephrase this comment to describe what the code does and not what the developer does?

kowshik-sundararajan commented on README.adoc:
```diff
@@ -1,11 +1,9 @@
 = Address Book (Level 4)
```
Remember to update the name of your app - it is not an Address Book anymore!

kowshik-sundararajan commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103W14-B4/addressbook-level4/tree/master
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Team name should be updated as well.

kowshik-sundararajan reviewed (commented):
Good work on updating user stories, use cases and other NFRs 👍 

Please check Travis to check why the build is failing and make the necessary changes to ensure that the build passes. 

Some general feedback:
* Good to see that you have created issues and `v1.0` milestone. Please remember to add `v1.1` to `v1.5` milestones by next tutorial - they can be empty but should be created with deadlines.

* Remember to update the target user profile and the description of the app.

* Do set up the issue tracker in your repo. 

* I see that individual PRs that didn't pass Travis have been merged in your repo - in practice, this shouldn't be acceptable. By merging pull requests that do not pass Travis, you will find it difficult to figure out why the main build fails.

*Do not close this PR!*

muruges95 commented:
Can I just check, do we rename the pull request to v1.1 for this coming  week?

muruges95 commented:
The reason why the build is failing is because of a style issue on upstream's version of the code; specifically in the file XMLAddressBookStorage.java .Locally we dont have any travis build fails, if you go to our repo page, you can see that both the AppVeyor and Travis builds are fine. 

![image](https://user-images.githubusercontent.com/14848271/37347473-de67264e-270c-11e8-8a45-fe1549d43764.png)
The above is an image from the travis for the check of this pull request. I notice that no other team is facing this issue, so would like to know what we did wrong and how to resolve this.

kowshik-sundararajan commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,16 +945,461 @@ Use case ends.
-_{More to be added}_
+=== Use case: Auto-fill
+
+*MSS*
```
Remember to update all references to `AddressBook` to your application's name!

muruges95 replied:
Noted

kowshik-sundararajan commented on docs/UserGuide.adoc:
```diff
@@ -11,13 +11,13 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4
+:repoURL: https://github.com/CS2103W14-B4/addressbook-level4
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Remeber to update the `By` (similar to Dev Guide).

muruges95 replied:
Noted

kowshik-sundararajan reviewed (commented):
Good job on updating the documentation.

@muruges95  @WJY-norainu  @Damienskt  @tzerbin 
Please indicate which features appear in which milestone. Also If there is something that is left for future work, need to indicate it as coming in v2.0.

I would want you to mention the following in your **Developer Guide**:

* For each team member what are the features (major and minor) you are proposing?
* Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

General feedback: 
* Include features and their description that are planned to be implemented in `v2.0` in the User Guide.
* Remember to replace all references to AddressBook 😛 
* You can be more specific in your user stories / dev guide - replacing person with celebrity. Ex: Add a celebrity, etc. 
* If you want me to review your code for quality checks, tag me in the PRs submitted to your repo.

*Note for next week*: Each member of the team should have at least 1 merged PR of their enhancement as `v1.2` release. (It doesn't have to be perfect code, but something that works).

As a favour, could you add a comment every weekend to summarize what are the changes that have been made? This would make it easier for me to review the changes.

kowshik-sundararajan reviewed (commented):
@muruges95 @WJY-norainu @Damienskt @tzerbin 
Since I've reviewed the DG and your enhancements in the tutorial, there's not much to add on here.

For your UG, good that you have mentioned which features were implemented in `v1.2` and which features will be implemented in `v1.3`. You can also specify which features will be implemented for the remaining milestones. (Basically, all the features should be present and they should have a milestone specified for each of them).

Damienskt committed:
Merge pull request #141 from Damienskt/Documentation-guide

updated Documentation guide

WJY-norainu committed:
Change viewDate input date format
Can read in dd-mm-yyyy or dd-mm
addAppointment will switch to the start date of appointment
Format the message of editAppointment

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Fix style violation

WJY-norainu committed:
Update UG for viewDate

WJY-norainu committed:
Create test for ViewDateCommandParser

WJY-norainu committed:
Refactor Model#deleteAppointment, Model#getChosenAppointment
Instead of throwing CommandException, throw IndexOutOfBoundsException to ensure that CommandException is only thrown in Logic component

WJY-norainu committed:
Fix style violation and update affected class

WJY-norainu committed:
Create test for viewDate

WJY-norainu committed:
Delete uminplemented test

WJY-norainu committed:
Merge pull request #142 from WJY-norainu/master

Change viewDate feature's input format...

muruges95 committed:
edited DG

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main
Keep local branch up to date

Damienskt committed:
included GoogleWebService class

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into EstimateRoute

Damienskt committed:
update UI

Damienskt committed:
add Api key

Damienskt committed:
update UI

tzerbin committed:
Updated UG docx

tzerbin committed:
Merge pull request #147 from tzerbin/week12-documentationmodification

Updated UG docx

tzerbin committed:
Fix hyperlink issues

tzerbin committed:
Merge pull request #148 from tzerbin/week12-documentationmodification

Fix hyperlink issues for UG

WJY-norainu committed:
Change listAppointment
allows user to specify a range from which appointments are listed
change static attributes and methods in ListAppointmentCommand
use currentDisplayedAppointments in ModelManager to store the information instead

WJY-norainu committed:
Fix style violation

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Merge pull request #149 from WJY-norainu/master

Extend functionality of listAppointment

WJY-norainu committed:
Create test for DeleteAppointmentCommand

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Merge pull request #150 from WJY-norainu/master

Create test for DeleteAppointmentCommand

Damienskt committed:
resolve style issues

Damienskt committed:
Merge pull request #155 from Damienskt/EstimateRoute

added GoogleWebService class

WJY-norainu committed:
Create test for ViewCalendarCommandParser and ViewCalendarByCommandParser

WJY-norainu committed:
no message

WJY-norainu committed:
Merge pull request #157 from WJY-norainu/master

Create test for ViewCalendarCommandParser and ViewCalendarByCommandParser

tzerbin committed:
Update compressed DG docx

tzerbin committed:
Merge pull request #158 from tzerbin/week12-documentationmodification

Update compressed DG docx

WJY-norainu committed:
Annotate

WJY-norainu committed:
Create test for ViewCalendarCommand

WJY-norainu committed:
Fix style violation

WJY-norainu committed:
Create test for ViewCalendarByCommand

WJY-norainu committed:
Create test for ViewCombinedCalendarCommand

WJY-norainu committed:
Create test for ViewDateCommand

WJY-norainu committed:
Fix style violation

WJY-norainu committed:
Remove equals test as it is not overriden in ViewCombinedCalendarCommand

WJY-norainu committed:
Merge pull request #166 from WJY-norainu/master

Create test

muruges95 committed:
Added addappointmentcommandtest

muruges95 committed:
fixed checkstyle checks

Damienskt committed:
Add viewappointment test

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Tests

Damienskt committed:
update addressbookParser test

Damienskt committed:
update viewAppointment test

Damienskt committed:
update tests

muruges95 committed:
Created Edit appointment command test

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into create-tests
Keep local branch up to date

Damienskt committed:
change position

Damienskt committed:
resolving travis and appveyor issue for viewappointmentSystem test

Damienskt committed:
resolve travis appveyor issue

Damienskt committed:
resolve appveyor and travis issue

muruges95 committed:
Merge pull request #167 from Damienskt/Tests

ViewAppointmentCommand test

muruges95 committed:
created editappointment parser test

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into create-tests
Keep local branch up to date

muruges95 committed:
fixed error in UG

muruges95 committed:
fixed bug in delete appointment test

muruges95 committed:
fixed some buggy tests

muruges95 committed:
fixed another buggy test

muruges95 committed:
added statement to buggy test

muruges95 committed:
fixed bugs in certain tests

muruges95 committed:
Merge pull request #168 from muruges95/create-tests

Create tests

Damienskt committed:
update view appointment system tests

Damienskt committed:
debug viewappointmentsystem test

Damienskt committed:
remove unused import

Damienskt committed:
debug viewApptSysTest

Damienskt committed:
remove unused import

Damienskt committed:
debug viewappt sys test

Damienskt committed:
remove unused imports

WJY-norainu committed:
Update listAppointment in UG

WJY-norainu committed:
Add project portfolio

Damienskt committed:
Update ugdg
Add Project portfolio

WJY-norainu committed:
Update project portfolio

WJY-norainu committed:
Add line breaks

Damienskt committed:
update map related tests

Damienskt committed:
Merge pull request #169 from Damienskt/Documentation-guide

Update UGDG diagrams and added project portfolio

Damienskt committed:
remove unused imports

Damienskt committed:
import order issue

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Tests

Damienskt committed:
debug tests

Damienskt committed:
Merge pull request #170 from Damienskt/Tests

Map related test update

muruges95 committed:
fixed some bugs with adding celeb to calendar

muruges95 committed:
removed time based test due to inconsistencies

tzerbin committed:
re-add back tests in serializablestoragecalendartest

tzerbin committed:
correct tests for XmlAdaptedAppointmentTest.java

tzerbin committed:
Correct checkstyle errors

tzerbin committed:
Merge pull request #171 from tzerbin/week12-storagetest

Week12 storage tests

tzerbin committed:
Revert "Week12 storage tests"

tzerbin committed:
Merge pull request #172 from CS2103JAN2018-W14-B4/revert-171-week12-storagetest

Revert "Week12 storage tests"

tzerbin committed:
Remove unnecessary code

tzerbin committed:
add author tag

tzerbin committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into week12-storagetest

tzerbin committed:
correct tags with certain errors

tzerbin committed:
Merge pull request #173 from tzerbin/week12-storagetest

Week12 storagetest

muruges95 committed:
removed coverall icon

muruges95 committed:
re-added coverall status

tzerbin committed:
update storage class diagram

tzerbin committed:
add readstorage sequence diagram

muruges95 committed:
added project portfolio

WJY-norainu committed:
Rename project portfolio

muruges95 committed:
Merge pull request #174 from muruges95/master

Added project portfolio

tzerbin committed:
update DG

tzerbin committed:
add PPP

WJY-norainu committed:
Merge pull request #175 from WJY-norainu/master

Update UG and project portfolio

tzerbin committed:
Remove whitespaces

muruges95 committed:
Merge pull request #176 from tzerbin/week12-sequencediagram

Update DG and PPP

tzerbin committed:
populate addressbook and storagecalendar xml

tzerbin committed:
Merge pull request #177 from tzerbin/week12-population

populate addressbook and storagecalendar xml, remove /data/ from .gitignore

WJY-norainu committed:
Remove the change of calendar when selecting a celebrity
Allows this will mess up viewCalendar's functionality

WJY-norainu committed:
Add test for removing celebrity tag
change javadoc

WJY-norainu committed:
Merge pull request #178 from WJY-norainu/master

Remove change of calendar when selecting celebrity

WJY-norainu committed:
Update UG for viewDate and viewCalendarBy

WJY-norainu committed:
Change top left text to CelebManager
Fix style violation

Damienskt committed:
update code quality

Damienskt committed:
resolve format issue

Damienskt committed:
Merge pull request #180 from Damienskt/master

update code style and format

muruges95 committed:
Included system test for add appointment command

Damienskt committed:
add image

Damienskt committed:
add manual testing for er sl and va

Damienskt committed:
typo error resolve

WJY-norainu committed:
Update UG

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Update UG
standardise figure labels

WJY-norainu committed:
Fix style violation
add a new line at EOF

muruges95 committed:
Added support for preventing duplicate appointments

WJY-norainu committed:
Update UG
specify base date

muruges95 committed:
fixed bugs in tests and addappointment

muruges95 committed:
fixed bugs in tests

muruges95 committed:
fixed typo

muruges95 committed:
Merge pull request #181 from muruges95/system-tests

Included system test for add appointment command

muruges95 committed:
included minimum of 15 for each appointment

tzerbin committed:
Update DG with pdf version

tzerbin committed:
Merge pull request #182 from tzerbin/week13-devguidemodifications

Update DG

tzerbin committed:
Amend tabs error

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Documentation-guide

Damienskt committed:
Merge pull request #183 from tzerbin/week13-devguidemodifications

Amend tabs error

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Documentation-guide

Damienskt committed:
correct naming of image

Damienskt committed:
Update sequence diagram for er and va

Damienskt committed:
update diagrams and project portfolio

tzerbin committed:
Remove highlights

tzerbin committed:
Merge pull request #184 from tzerbin/week13-devguidemodifications

Remove highlights in DG

muruges95 committed:
added to addappointment system test

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into system-tests
Keep local branch up to date

WJY-norainu committed:
Update UG

muruges95 committed:
removed data folder

muruges95 committed:
Update config.json

Add newline

muruges95 committed:
Merge pull request #179 from WJY-norainu/master

Update UG

muruges95 committed:
resolve conflict

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Documentation-guide

Damienskt committed:
update json

Damienskt committed:
add product image

WJY-norainu committed:
Update DG Manual Testing

Add testing instructions for removeTag, deleteAppointment, viewCalendar, viewCombinedCalendar, viewCalendarBy and viewDate commands

Damienskt committed:
new end of line for rof

muruges95 committed:
Completely fixed add appointment command system test

Damienskt committed:
change Api key

muruges95 committed:
removed config file from tracking

muruges95 committed:
Merge pull request #186 from muruges95/system-tests

Add appointment command system tests

Damienskt committed:
Merge branch 'master' into Documentation-guide

Damienskt committed:
Update GoogleWebServices.java

Damienskt committed:
update googlewebservices

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main into Documentation-guide

# Conflicts:
#	config.json
#	docs/DeveloperGuide.adoc
#	src/main/java/seedu/address/logic/map/GoogleWebServices.java

WJY-norainu committed:
Replace pics for viewAppointment

WJY-norainu committed:
Merge branch 'master' of https://github.com/WJY-norainu/main

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

Damienskt committed:
Merge branch 'Documentation-guide' of https://github.com/Damienskt/addressbook-level4 into Documentation-guide

# Conflicts:
#	docs/DeveloperGuide.adoc
#	src/main/java/seedu/address/logic/map/GoogleWebServices.java

Damienskt committed:
hide saving data manual testing

Damienskt committed:
add image for about us

Damienskt committed:
add nelwine at eof

Damienskt committed:
Merge pull request #187 from Damienskt/Documentation-guide

update documentation

WJY-norainu committed:
Update UG - change command summary to table form
Add explanation for removeTag

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

Damienskt committed:
update UiComponentClassDiagram

Damienskt committed:
update UiComponentClassDiagram

Damienskt committed:
remove old uiClassDiagram

Damienskt committed:
remove old uiComponentClassDiagram

Damienskt committed:
update new uiClassDiagram

Damienskt committed:
Update UiClassDiagram

WJY-norainu committed:
Add figures to exaplain removeTag

WJY-norainu committed:
Add pic to explain removeTag

WJY-norainu committed:
Change arrow size in figures

WJY-norainu committed:
Update DG
Add manual testing for listAppointment
Change ordering of commands in manual testing

WJY-norainu committed:
Merge pull request #188 from WJY-norainu/master

Update UGDG

tzerbin committed:
Refine DG

tzerbin committed:
Merge pull request #189 from tzerbin/week13-devguidemodifications

Refine DG

WJY-norainu committed:
Update DG for removeTag and deleteAppointment
change sequence diagrams for removeTag and deleteAppointment commands

WJY-norainu committed:
Update 2101 UG

WJY-norainu committed:
Merge pull request #190 from WJY-norainu/master

Update UGDG

tzerbin committed:
Amend read storage sequence diagram

tzerbin committed:
Update PPP

tzerbin committed:
Add manual testing in DG.adoc appendix F

tzerbin committed:
Update sequence diagram for DG

tzerbin committed:
update PPP

tzerbin committed:
update DG Appendix F

Damienskt committed:
update dg and project portfolio

Damienskt committed:
Merge pull request #194 from Damienskt/master

update dg and project portfolio

muruges95 committed:
included manual testing for edit appointment

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main
Keep local branch up to date

muruges95 committed:
included annotations

muruges95 committed:
Merge branch 'master' into master

muruges95 committed:
Merge pull request #193 from tzerbin/master

Update documentations

Damienskt committed:
update naming of ppp

Damienskt committed:
update annotation to collate files

Damienskt committed:
Merge branch 'master' of https://github.com/Damienskt/addressbook-level4

Damienskt committed:
Merge pull request #195 from Damienskt/master

Annotations for collating code

tzerbin committed:
Update PPP

tzerbin committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

tzerbin committed:
Update AboutUs.adoc with links to team's project portfolio

tzerbin committed:
Edit typo

muruges95 committed:
updated ppp

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main
Keep local branch up to date

tzerbin committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

tzerbin committed:
Merge pull request #196 from tzerbin/master

Update documentations (PPP, AboutUs)

WJY-norainu committed:
Replace Model Class Diagram

Damienskt committed:
update GoogleWebServices

Damienskt committed:
change Api key

Damienskt committed:
update comments for api key

WJY-norainu committed:
Update Model component description

Damienskt committed:
Merge branch 'master' of https://github.com/WJY-norainu/main

Damienskt committed:
remove trailing whitespaces

Damienskt committed:
Merge pull request #197 from Damienskt/master

Documentation update

Damienskt committed:
remove trailing whitespaces

Damienskt committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Annotate

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

muruges95 committed:
modified ppp

muruges95 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main
Keep local branch up to date

muruges95 committed:
change api key

WJY-norainu committed:
Remove word doc and update DG

WJY-norainu committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B4/main

WJY-norainu committed:
Upload collated codes

WJY-norainu committed:
Update PPP

WJY-norainu committed:
Update PPP

WJY-norainu committed:
Update PPP

Damienskt committed:
update PPP and api key

WJY-norainu committed:
Change formatting and hyperlink in PPP

WJY-norainu committed:
Update Sequence Diagram - labels

WJY-norainu committed:
Update UGDG

muruges95 committed:
made final edit to docs

WJY-norainu committed:
Update DG
replace URL
update Use Case

WJY-norainu committed:
Update DG - link for MainWindow.java and MainWindow.fxml

WJY-norainu committed:
Updated PPP

muruges95 committed:
Merge branch 'master' of https://github.com/WJY-norainu/main
Keep local branch up to date

muruges95 committed:
changed API key being used
